packages/gatsby/src/redux/run-sift.js

```diff
@@ -163,15 +192,21 @@ const getBucketsForFilters = (filters, nodeTypeNames, typedKeyValueIndexes) => {
       query: { value: targetValue },
     } = filter
 
-    ensureIndexByTypedChain(chain, nodeTypeNames, typedKeyValueIndexes)
+    let cacheKey = createTypedFilterCacheKey(nodeTypeNames, filter)
```

Comment:
cacheKey recursively collapses elemMatch chain, but chain here only takes original path.

Reply:
`elemMatch` does not reach this place, yet.

In my `elemMatch` implementation this function is called with a full path, where it has walked all the `nestedQuery` steps of an `elemMatch` filter. Same argument types.

Reply:
Added a comment.

Thanks :)